lib/node_modules/@stdlib/math/base/special/chebyshev-t-polynomial/README.md

```diff
@@ -0,0 +1,154 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2025 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# chebyshevtpoly
+
+> [Chebyshev Polynomial][chebyshev-polynomial] of the first kind.
+
+<section class="intro">
+
+The [Chebyshev Polynomial][chebyshev-polynomial] of the first kind is defined as
+
+<!-- <equation class="equation" label="eq:chebyshev_polynomial" align="center" raw="T_n(x) = \cos( n \cos^{-1}(x) )" alt="Chebyshev Polynomial of the first kind."> -->
+
+```math
+T_n(x) = \cos( n \cos^{-1}(x) )
+```
+
+<!-- </equation> -->
+
+where `n` is the polynomial degree.
+
+The [Chebyshev Polynomial][chebyshev-polynomial] of the first kind is related to the Gaussian hypergeometric function via the following equation
+
+<!-- <equation class="equation" label="eq:chebyshev_polynomial2" align="center" raw="T_n(x) = {}_2F_1\left( -n, n; \frac{1}{2}; \frac{1}{2}(1-x) \right)" alt="Chebyshev Polynomial of the first kind expressed in terms of the Gaussian hypergeometric function."> -->
+
+```math
+T_n(x) = {}_2F_1\left( -n, n; \frac{1}{2}; \frac{1}{2}(1-x) \right)
+```
+
+<!-- </equation> -->
+
+</section>
+
+<!-- /.intro -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var chebyshevtpoly = require( '@stdlib/math/base/special/chebyshev-t-polynomial' );
+```
+
+#### chebyshevtpoly( n, x )
+
+Evaluates the [Chebyshev Polynomial][chebyshev-polynomial] of the first kind for a degree `n` at a value `x`.
+
+```javascript
+var y = chebyshevtpoly( 0.0, 0.5 );
+// returns 1.0
+
+y = chebyshevtpoly( 1.0, -0.5 );
+// returns -0.5
+
+y = chebyshevtpoly( 5.0, 0.5 );
+// returns 0.5
+```
+
+If provided `NaN` as any argument, the function returns `NaN`.
+
+```javascript
+var y = chebyshevtpoly( NaN, 1.0 );
+// returns NaN
+
+y = chebyshevtpoly( 0.0, NaN );
+// returns NaN
+```
+
+If provided a polynomial degree `n < 0`, the function returns `NaN`.
+
+```javascript
+var y = chebyshevtpoly( -2.0, 0.5 );
+// returns NaN
+
+y = chebyshevtpoly( -4.5, -0.5 );
+// returns NaN
+```
+
+If provided a polynomial degree `n` which is not a nonnegative integer, the function returns `NaN`.
```

Comment:
Why do we restrict to only integer degrees? Why not follow SciPy and state that non-integer values are resolved relative to the hypergeometric function?